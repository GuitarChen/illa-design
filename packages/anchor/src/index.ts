export * from "./anchor"
export * from "./interface"
