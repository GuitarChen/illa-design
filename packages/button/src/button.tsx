import { Children, forwardRef, useMemo } from "react"
import { ButtonProps } from "./interface"
import { css } from "@emotion/react"
import { LoadingIcon } from "@illa-design/icon"
import {
  applyBg,
  applyCursor,
  applyElementColor,
  applyFontStyle,
  applyIconWithoutText,
  applyLeftIconStyle,
  applyPaddingStyle,
  applyRightIconStyle,
  applyShape,
  applyTagContainer,
  applyWithoutTextSize,
} from "./style"
import { ButtonGroupContext } from "."
import { getSizeCssByAutoFullProps } from "@illa-design/system"

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    const { autoFullHorizontal, autoFullVertically } = props
    const sizeCss = useMemo(
      () => getSizeCssByAutoFullProps(autoFullHorizontal, autoFullVertically),
      [autoFullHorizontal, autoFullVertically],
    )

    return (
      <ButtonGroupContext.Consumer>
        {(value) => {
          const { attached, first, last } = value ?? {}

          const {
            _css,
            colorScheme = value?.colorScheme ?? "blue",
            size = value?.size ?? "medium",
            variant = value?.variant ?? "fill",
            shape = value?.shape ?? "square",
            fullWidth,
            loading,
            loadingText,
            leftIcon,
            disabled,
            rightIcon,
            buttonRadius,
            borderColor,
            backgroundColor,
            textColor,
            autoFullHorizontal,
            autoFullVertically,
            ...otherProps
          } = props

          const hasLoadingText =
            loadingText != undefined && loadingText.length > 0
          const hasChild = Children.count(props.children) >= 1
          const hasPropContent = hasChild || (hasLoadingText && loading)
          const finalContainer = css`
            ${applyTagContainer(fullWidth)};
            ${applyCursor(loading ?? false, disabled ?? false)}
            ${hasPropContent
              ? applyPaddingStyle(size, variant)
              : applyWithoutTextSize(size, fullWidth)};
            ${applyShape(
              shape,
              attached ?? false,
              first ?? false,
              last ?? false,
            )};
            ${applyElementColor(variant, colorScheme, textColor)};
            ${applyBg(variant, colorScheme, backgroundColor, borderColor)};
            ${buttonRadius ? `border-radius: ${buttonRadius};` : ""}
          `

          return (
            <button
              ref={ref}
              css={css(finalContainer, sizeCss, _css)}
              disabled={disabled || loading}
              {...otherProps}
            >
              {(loading || leftIcon) && (
                <span
                  css={
                    hasPropContent
                      ? applyLeftIconStyle(size)
                      : applyIconWithoutText(size)
                  }
                >
                  {loading ? <LoadingIcon spin={true} /> : leftIcon}
                </span>
              )}
              {hasPropContent && (
                <span css={applyFontStyle(size)}>
                  {loading && loadingText ? loadingText : props.children}
                </span>
              )}
              {rightIcon && (
                <span
                  css={
                    hasPropContent
                      ? applyRightIconStyle(size)
                      : applyIconWithoutText(size)
                  }
                >
                  {rightIcon}
                </span>
              )}
            </button>
          )
        }}
      </ButtonGroupContext.Consumer>
    )
  },
)

Button.displayName = "Button"
