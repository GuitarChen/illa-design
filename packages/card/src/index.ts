export * from "./card"
export * from "./interface"
export * from "./card-grid"
export * from "./meta"
