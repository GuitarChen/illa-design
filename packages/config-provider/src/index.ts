export * from "./config-provider"
export * from "./interface"
export * from "./config-provider-context"
export * from "./config"

// locale
export * from "./locale/en-US"
export * from "./locale/zh-CN"
export * from "./locale/def"
