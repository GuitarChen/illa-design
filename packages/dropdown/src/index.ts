export * from "./dropdown"
export * from "./interface"
