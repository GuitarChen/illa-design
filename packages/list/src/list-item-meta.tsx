import { forwardRef } from "react"
import { ListItemMetaProps } from "./interface"
import {
  applyItemMetaAvatar,
  applyItemMetaContainer,
  applyItemMetaTitle,
  applyTypoStyle,
} from "./style"
import { Avatar } from "@illa-design/avatar"
import { Paragraph, Text, Typography } from "@illa-design/typography"
import { globalColor, illaPrefix } from "@illa-design/theme"

export const ListItemMeta = forwardRef<HTMLDivElement, ListItemMetaProps>(
  (props, ref) => {
    const { title, avatar, description, ...otherProps } = props
    return (
      <div css={applyItemMetaContainer} ref={ref} {...otherProps}>
        {avatar && (
          <Avatar
            css={applyItemMetaAvatar}
            size="medium"
            shape="square"
            src={avatar}
          />
        )}
        <Typography css={applyTypoStyle}>
          {title && (
            <Text fontSize="14px" colorScheme="gray" bold>
              {title}
            </Text>
          )}
          {title && description && <div style={{ height: "4px" }} />}
          {description && (
            <Paragraph
              colorScheme={
                title == undefined
                  ? globalColor(`--${illaPrefix}-grayBlue-02`)
                  : globalColor(`--${illaPrefix}-grayBlue-04`)
              }
              fontSize="14px"
            >
              {description}
            </Paragraph>
          )}
        </Typography>
      </div>
    )
  },
)

ListItemMeta.displayName = "ListItemMeta"
