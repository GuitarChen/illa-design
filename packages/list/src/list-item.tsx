import { forwardRef } from "react"
import { css } from "@emotion/react"
import { ListItemProps } from "./interface"
import {
  applyListItemActionsStyle,
  applyListItemContainer,
  applyListItemExtraStyle,
  applyListItemInner,
} from "./style"

export const ListItem = forwardRef<HTMLDivElement, ListItemProps>(
  (props, ref) => {
    const { actions, extra, _css, ...otherProps } = props
    return (
      <div css={css(applyListItemContainer, _css)} ref={ref} {...otherProps}>
        <div css={applyListItemInner}>
          {props.children}
          {actions && <div css={applyListItemActionsStyle}>{actions}</div>}
        </div>
        {extra && <div css={applyListItemExtraStyle}>{extra}</div>}
      </div>
    )
  },
)

ListItem.displayName = "ListItem"
