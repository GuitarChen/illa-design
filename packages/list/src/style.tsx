import { css, SerializedStyles } from "@emotion/react"
import { globalColor, illaPrefix } from "@illa-design/theme"
import { ListSize } from "./interface"

export function applyListContainer(bordered?: boolean): SerializedStyles {
  let containerCss = css`
    display: inline-flex;
    flex-direction: column;
  `
  if (bordered) {
    containerCss = css`
      ${containerCss};
      border: solid 1px ${globalColor(`--${illaPrefix}-grayBlue-09`)};
      border-radius: 2px;
    `
  }
  return containerCss
}

export function applyListItemOuter(
  size: ListSize,
  hoverable?: boolean,
): SerializedStyles {
  let sizeCss: SerializedStyles
  switch (size) {
    case "small":
      sizeCss = css`
        padding: 8px 16px;
      `
      break
    case "medium":
      sizeCss = css`
        padding: 12px 16px;
      `
      break
    case "large":
      sizeCss = css`
        padding: 16px 16px;
      `
      break
  }
  let hoverableCss: SerializedStyles = css``
  if (hoverable) {
    hoverableCss = css`
      &:hover {
        background-color: ${globalColor(`--${illaPrefix}-grayBlue-09`)};
      }
    `
  }
  return css`
    ${sizeCss};
    ${hoverableCss};
  `
}

export const applyListItemContainer = css`
  display: flex;
  flex-direction: row;
`

export const applyListItemInner = css`
  display: inline-flex;
  flex-direction: column;
  flex-grow: 1;
`

export const applyItemMetaContainer = css`
  display: inline-flex;
  align-items: start;
  flex-direction: row;
`

export const applyItemMetaInner = css`
  display: inline-flex;
  flex-direction: row;
`

export const applyItemMetaAvatar = css`
  margin-right: 8px;
`

export const applyTypoStyle = css`
  align-self: center;
`

export const applyItemMetaTitle = css`
  margin-bottom: 4px;
`

export const applyListItemActionsStyle = css`
  margin-top: 8px;
`

export const applyListItemExtraStyle = css`
  margin-left: 32px;
`

export function applyBarStyle(size: ListSize) {
  let sizeCss: SerializedStyles
  switch (size) {
    case "small":
      sizeCss = css`
        padding: 8px 16px;
      `
      break
    case "medium":
      sizeCss = css`
        padding: 12px 16px;
      `
      break
    case "large":
      sizeCss = css`
        padding: 16px 16px;
      `
      break
  }
  return css`
    ${sizeCss};
  `
}
