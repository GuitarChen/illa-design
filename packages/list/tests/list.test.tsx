import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import { List, ListItem, ListItemMeta } from "../src"
import { Button } from "@illa-design/button"
import { Image } from "@illa-design/image"

test("List renders header and footer.", () => {
  render(
    <List
      data-testid="test-list"
      data={[
        { title: "Title A", description: "Desc A" },
        { title: "Title B", description: "Desc B" },
        { title: "Title C", description: "Desc C" },
        { title: "Title D", description: "Desc D" },
        { title: "Title E", description: "Desc E" },
      ]}
      render={(data, index) => {
        return (
          <ListItem
            actions={<Button>Actions</Button>}
            extra={<Image src={"https://devbo.cn/logo.svg"} />}
          >
            <ListItemMeta title={data.title} description={data.description} />
          </ListItem>
        )
      }}
      renderKey={(data, index) => {
        return index.toString()
      }}
      header={<span>Header</span>}
      footer={<span>Footer</span>}
    />,
  )
  expect(screen.getByText("Header")).toBeInTheDocument()
  expect(screen.getByText("Footer")).toBeInTheDocument()
})

test("List renders without border and split", () => {
  render(
    <List
      data-testid="test-list"
      data={[
        { title: "Title A", description: "Desc A" },
        { title: "Title A", description: "Desc A" },
      ]}
      render={(data) => {
        return (
          <ListItem>
            <ListItemMeta title={data.title} description={data.description} />
          </ListItem>
        )
      }}
      renderKey={(data, index) => {
        return index.toString()
      }}
    />,
  )
  expect(screen.getByTestId("test-list")).toMatchSnapshot()
})
