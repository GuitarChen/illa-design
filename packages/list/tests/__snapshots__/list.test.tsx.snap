// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders without border and split 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: solid 1px #f2f3f5;
  border-radius: 2px;
}

.emotion-1 {
  padding: 12px 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-5 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-6 {
  word-break: break-all;
}

.emotion-6 h1 {
  margin: 0;
  font-size: 36px;
  line-height: normal;
}

.emotion-6 h2 {
  margin: 0;
  font-size: 32px;
  line-height: normal;
}

.emotion-6 h3 {
  margin: 0;
  font-size: 28px;
  line-height: normal;
}

.emotion-6 h4 {
  margin: 0;
  font-size: 24px;
  line-height: normal;
}

.emotion-6 h5 {
  margin: 0;
  font-size: 20px;
  line-height: normal;
}

.emotion-6 h6 {
  margin: 0;
  font-size: 16px;
  line-height: normal;
}

.emotion-6 p {
  margin: 0;
}

.emotion-7 {
  font-size: 14px;
}

.emotion-8 {
  color: #1f1f1f;
  font-weight: bold;
}

.emotion-9 {
  font-size: 14px;
}

.emotion-10 {
  color: #a9aeb8;
}

.emotion-11 {
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e5e6eb;
}

.emotion-11 span {
  padding: 0 16px;
}

.emotion-12 {
  vertical-align: middle;
  border-style: solid;
  border-width: 0 0 1px 0;
  width: 100%;
}

<div
  class="emotion-0"
  data-testid="test-list"
>
  <div
    class="rc-virtual-list"
    style="position: relative;"
  >
    <div
      class="rc-virtual-list-holder"
    >
      <div>
        <div
          class="rc-virtual-list-holder-inner"
          style="display: flex; flex-direction: column;"
        >
          <div
            class="emotion-1"
          >
            <div
              class="emotion-2"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-4"
                >
                  <article
                    class="emotion-5 emotion-6"
                  >
                    <span
                      class="emotion-7"
                    >
                      <span
                        style="position: relative;"
                      >
                        <iframe
                          style="background: transparent; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%; z-index: -1;"
                        />
                        <span
                          class="emotion-8"
                        >
                          Title A
                        </span>
                      </span>
                    </span>
                    <div
                      style="height: 4px;"
                    />
                    <div
                      class="emotion-9"
                    >
                      <span
                        style="position: relative;"
                      >
                        <iframe
                          style="background: transparent; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%; z-index: -1;"
                        />
                        <span
                          class="emotion-10"
                        >
                          Desc A
                        </span>
                      </span>
                    </div>
                  </article>
                </div>
              </div>
            </div>
          </div>
          <div
            class="emotion-11"
          >
            <div
              class="emotion-12"
            />
          </div>
          <div
            class="emotion-1"
          >
            <div
              class="emotion-2"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-4"
                >
                  <article
                    class="emotion-5 emotion-6"
                  >
                    <span
                      class="emotion-7"
                    >
                      <span
                        style="position: relative;"
                      >
                        <iframe
                          style="background: transparent; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%; z-index: -1;"
                        />
                        <span
                          class="emotion-8"
                        >
                          Title A
                        </span>
                      </span>
                    </span>
                    <div
                      style="height: 4px;"
                    />
                    <div
                      class="emotion-9"
                    >
                      <span
                        style="position: relative;"
                      >
                        <iframe
                          style="background: transparent; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%; z-index: -1;"
                        />
                        <span
                          class="emotion-10"
                        >
                          Desc A
                        </span>
                      </span>
                    </div>
                  </article>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
